.circleci/config.yml

```diff
@@ -0,0 +1,64 @@
+version: 2
+
+dry:
+  restore_shards_cache: &restore_shards_cache
+    keys:
+      - shards-cache-v1-{{ .Branch }}-{{ checksum "shard.lock" }}
+      - shards-cache-v1-{{ .Branch }}
+      - shards-cache-v1
+
+  save_shards_cache: &save_shards_cache
+    key: shards-cache-v1-{{ .Branch }}-{{ checksum "shard.lock" }}
+    paths:
+      - ./shards-cache
+
+  prepare: &prepare
+    command: |
+      make
+      git config --global user.email "you@example.com"
+      git config --global user.name "Your Name"
+      git config --global column.ui always
+
+jobs:
+  test:
+    docker:
+      - image: crystallang/crystal:latest
+        environment:
+          SHARDS_CACHE_PATH: ./shards-cache
+    steps:
+      - run: crystal --version
+      - checkout
+      - run: *prepare
+      - restore_cache: *restore_shards_cache
+      - run: shards
+      - save_cache: *save_shards_cache
+      - run: make test
+
+  test-on-nightly:
+    docker:
+      - image: crystallang/crystal:nightly
+        environment:
+          SHARDS_CACHE_PATH: ./shards-cache
+    steps:
+      - run: crystal --version
+      - checkout
+      - run: *prepare
+      - restore_cache: *restore_shards_cache
+      - run: shards
+      - run: make test
+
+workflows:
+  version: 2
+  ci:
+    jobs:
+      - test
+  nightly:
+    triggers:
+      - schedule:
+          cron: "0 0 * * *"
```

Comment:
I'm sure circle would appreciate it if you randomized these values instead of everyone submitting jobs at exactly midnight.